File: src/service/ModelBuilder.Models/Application/ConnectionAm.cs
using Mb.Models.Client;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Mb.Models.Application;

#region ConnectionAm

public class ConnectionAm : IValidatableObject
{
    [Required]
    public Guid Id { get; set; }
    [Required]
    public string FromConnector { get; set; }
    [Required]
    public string ToConnector { get; set; }
    [Required]
    public Guid MainProject { get; set; }
    [Required]
    public Guid Project { get; set; }

    public ICollection<HandleAm> Handles { get; set; }

    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
    {
        return new List<ValidationResult>();
    }
}

#endregion ConnectionAm

#region ConnectionTerminalAm

public class ConnectionTerminalAm : ConnectionAm
{
    public string TerminalType { get; set; }
    public string TerminalParentType { get; set; }
}

#endregion ConnectionTerminalAm

#region ConnectionRelationAm

public abstract class ConnectionRelationAm : ConnectionAm
{
}

#endregion ConnectionRelationAm

#region ConnectionFulfilledByAm

public class ConnectionFulfilledByAm : ConnectionRelationAm
{
}

#endregion ConnectionFulfilledByAm

#region ConnectionHasLocationAm

public class ConnectionHasLocationAm : ConnectionRelationAm
{
}

#endregion ConnectionHasLocationAm

#region ConnectionPartOfAm

public class ConnectionPartOfAm : ConnectionRelationAm
{
}

#endregion ConnectionPartOfAm

File: src/client/src/lib/classes/application/ConnectionAm.ts
import {
  Connection,
  ConnectionFulfilledBy,
  ConnectionHasLocation,
  ConnectionPartOf,
  ConnectionRelation,
  ConnectionTerminal,
} from "../Connection";

export class ConnectionAm {
  id: string;
  project: string;
  fromConnector: string;
  toConnector: string;
  mainProject: string;

  public constructor(obj: Connection) {
    this.id = obj.id;
    this.project = obj.project;
    this.fromConnector = obj.fromConnector;
    this.toConnector = obj.toConnector;
    this.mainProject = obj.mainProject;
  }
}

export class ConnectionTerminalAm extends ConnectionAm {
  public constructor(obj: ConnectionTerminal) {
    super(obj);
  }
}

export class ConnectionRelationAm extends ConnectionAm {
  public constructor(obj: ConnectionRelation) {
    super(obj);
  }
}

export class ConnectionFulfilledByAm extends ConnectionRelationAm {
  public constructor(obj: ConnectionFulfilledBy) {
    super(obj);
  }
}

export class ConnectionHasLocationAm extends ConnectionRelationAm {
  public constructor(obj: ConnectionHasLocation) {
    super(obj);
  }
}

export class ConnectionPartOfAm extends ConnectionRelationAm {
  public constructor(obj: ConnectionPartOf) {
    super(obj);
  }
}
